### ISSUE TITLE
Results are not deterministic in eval mode

### ISSUE THREAD

**Issue by arti32lehtonen** [lucidrains/performer-pytorch#10]
Hi!

It seems that results even in the eval mode are not deterministic at all. It can be explained by the kernel construction procedure described in the paper. But is it a way to get deterministic results?

I use PerformerLm with following parameters:
```
    self.performer = PerformerLM(
        num_tokens=16000,
        max_seq_len=512,
        dim=4,
        depth=4,
        heads=4,
        causal=True,
        nb_features=None,
        generalized_attention=False,
        reversible=False,
    )
```

**Comment by lucidrains** [comment]
@arti32lehtonen not without turning off the redrawing of weights, which hurts performance, as indicated in the figures in the pa[er

**Comment by lucidrains** [comment]
@arti32lehtonen i suppose one could turn off the redrawing of weights only at the end of training though :thinking:

**Comment by arti32lehtonen** [comment]
Yep, it helps. I do it after training as you suggest:
```
for layer in model.performer.performer.net.layers:
    layer[0].fn.fast_attention.redraw_projection = False
    projection_matrix = layer[0].fn.fast_attention.create_projection()
    layer[0].fn.fast_attention.register_buffer('projection_matrix', projection_matrix)
```

**Comment by lucidrains** [comment]
@arti32lehtonen addressed this in https://github.com/lucidrains/performer-pytorch/releases/tag/0.5.1
